state-sync/aptos-data-client/src/aptosnet/mod.rs

```diff
@@ -136,15 +140,24 @@ impl AptosNetDataClient {
         &self,
         request: &StorageServiceRequest,
     ) -> Result<PeerNetworkId, Error> {
-        let all_connected_peers = self.get_all_connected_peers()?;
-
-        // Identify the peers that can service this request
+        // Try to send the request to a priority peer that can service it
         let internal_peer_states = self.peer_states.read();
-        let serviceable_peers = all_connected_peers
+        let priority_connected_peers = self.get_priority_connected_peers()?;
```

Comment:
Practically, this won't have any difference today: both validators and VFNs (validator full nodes) are "prioritized" according to our heuristic, so validators and VFNs will pull data from each other as they already do. And PFNs (public full nodes) are only talking to 2 peers by default, so we shouldn't see any impact in devnet. The only reason for this change here is to ensure "subscription requests" are sent to prioritized peers (to help reduce latency). If we don't do this, it's possible a VFN will send the subscription request to a PFN (which is pointless, because they're downstream :P)

We'll definitely revisit this once we have topology information though. Unfortunately, we're in this really odd place today where we can't make smarter decisions because we don't have the topology data :(

Reply:
@JoshLind - Thanks for the explanation. Does it make sense to make this change only for subscription requests though?

Reply:
Also for subscription requests, do you think it makes sense for us to subscribe to multiple peers rather than just one to reduce the potential latency from subscribing to one slow peer?

Reply:
@sitalkedia, yeah, that's what I'm thinking. My general approach here is to iterate and make things better, focusing on the biggest fires first. If it wasn't for the cut, I'd delay this. But, then again, it's already 1k+ lines of code, so it's probably as big as it's gonna get for a single unit of work 😄